app/src/main/java/com/celzero/bravedns/service/BraveVPNService.kt

```diff
@@ -729,99 +777,82 @@ class BraveVPNService : VpnService(), ConnectionCapabilityMonitor.NetworkListene
-            Log.i(LOG_TAG, "$FILE_LOG_TAG EXCLUDE_FROM_VPN - restartVpn- ${appMode.getDNSMode()}")
-            restartVpn(appMode.getDNSMode(), appMode.getFirewallMode(), appMode.getProxyMode())
-        }
-
-        if (PersistentState.IS_SCREEN_OFF == key) {
-            isScreenLocked = persistentState.getScreenLockData()
-            Log.i(LOG_TAG, "$FILE_LOG_TAG preference for screen off mode is modified - $isScreenLocked")
-        }
-
-        if (PersistentState.BACKGROUND_MODE == key) {
-            isBackgroundEnabled = persistentState.backgroundEnabled && Utilities.isAccessibilityServiceEnabledEnhanced(this, BackgroundAccessibilityService::class.java)
-            Log.i(LOG_TAG, "$FILE_LOG_TAG preference for background mode is modified - $isBackgroundEnabled")
-        }
-        if (PersistentState.BLOCK_UNKNOWN_CONNECTIONS == key) {
-            Log.i(LOG_TAG, "$FILE_LOG_TAG preference for block unknown connections is modified")
-            blockUnknownConnection = persistentState.blockUnknownConnections
-        }
-
-        if (PersistentState.LOCAL_BLOCK_LIST == key) {
-            Log.i(LOG_TAG, "$FILE_LOG_TAG preference for local block list is changed - restart vpn")
-            spawnServerUpdate()
-        }
-
-        if (PersistentState.LOCAL_BLOCK_LIST_STAMP == key) {
-            Log.i(LOG_TAG, "$FILE_LOG_TAG configuration stamp for local block list is changed- restart vpn")
-            spawnServerUpdate()
-        }
-
-        if (PersistentState.ALLOW_BYPASS == key) {
-            Log.i(LOG_TAG, "$FILE_LOG_TAG preference for allow by pass is changed - restart vpn")
-            restartVpn(appMode.getDNSMode(), appMode.getFirewallMode(), appMode.getProxyMode())
-        }
-
-        if (PersistentState.PRIVATE_DNS == key) {
-            privateDNSOverride = persistentState.allowPrivateDNS
-        }
-
-        if (PersistentState.HTTP_PROXY_ENABLED == key) {
-            Log.i(LOG_TAG, "$FILE_LOG_TAG preference for http proxy is changed - restart vpn")
-            restartVpn(appMode.getDNSMode(), appMode.getFirewallMode(), appMode.getProxyMode())
-        }
-
-        if (PersistentState.BLOCK_UDP_OTHER_THAN_DNS == key) {
-            blockUDPTraffic = persistentState.udpBlockedSettings
-        }
-
-        if(PersistentState.ORBOT_MODE_CHANGE == key){
-            Log.i(LOG_TAG, "$FILE_LOG_TAG ORBOT proxy change - restart vpn ${appMode.getProxyMode()}")
-            restartVpn(appMode.getDNSMode(), appMode.getFirewallMode(), appMode.getProxyMode())
+            PersistentState.DNS_PROXY_ID -> {
+                restartVpn(appMode.getDNSMode(), appMode.getFirewallMode(), appMode.getProxyMode())
+            }
+            PersistentState.EXCLUDE_FROM_VPN -> {
+                restartVpn(appMode.getDNSMode(), appMode.getFirewallMode(), appMode.getProxyMode())
+            }
+            PersistentState.IS_SCREEN_OFF -> {
+                isScreenLocked = persistentState.isScreenOff
+            }
```

Comment:
If the `isScreenOff` is not persisted then the value is getting lost when the VPN is restarted. 
We were having it in-memory earlier and then changed to shared pref because of the issue.